crates/libs/sys/src/core/mod.rs

```diff
@@ -34,3 +34,33 @@ impl GUID {
         Self { data1: (uuid >> 96) as u32, data2: (uuid >> 80 & 0xffff) as u16, data3: (uuid >> 64 & 0xffff) as u16, data4: (uuid as u64).to_be_bytes() }
     }
 }
+
+#[doc(hidden)]
+#[macro_export]
+macro_rules! windows_link {
+    ($library:literal, $abi:literal fn $name:ident($($arg:ident: $argty:ty),*)->$ret:ty) => (
+        #[cfg(all(feature = "raw_dylib", target_arch = "x86"))]
```

Comment:
Maybe move the cfg's from the macro body to the macro definition. That is have 5 windows_link macros each producing only a single extern block with the cfg determining which macro is enabled. This should reduce the amount of code rustc has to churn through to a fifth of what it currently has to.

Reply:
Ah, good idea!